clients/unshaded/src/main/java/tachyon/hadoop/ConfUtils.java

```diff
@@ -73,9 +74,21 @@ public static TachyonConf loadFromHadoopConfiguration(Configuration source) {
         LOG.error("Unable to load TachyonConf from Hadoop configuration", e);
         throw new RuntimeException(e);
       }
-
-      return tachyonConfProperties != null ? new TachyonConf(tachyonConfProperties) : null;
     }
-    return null;
+    if (tachyonConfProperties == null) {
+      tachyonConfProperties = new Properties();
+    }
+    // Load any Tachyon configuration parameters existing in the Hadoop configuration.
+    for (Map.Entry<String, String> entry : source) {
+      String propertyName = entry.getKey();
+      // TODO: a better way to enumerate every Tachyon configuration parameter
+      if (propertyName.startsWith("tachyon.")
+          || propertyName.equals(Constants.S3_ACCESS_KEY)
+          || propertyName.equals(Constants.S3_SECRET_KEY)) {
+        tachyonConfProperties.put(propertyName, entry.getValue());
+      }
+    }
+    LOG.info("Loading Tachyon properties from Hadoop configuration: " + tachyonConfProperties);
```

Comment:
this is called on each client instance. Maybe this is fine so we could see the extracted Hadoop conf per task.